### DISCUSSION TITLE
Dynamically Pass lesson_category_id and Store Lessons with id and title in JSON Field in CRUD Form

### DISCUSSION THREAD

**Discussion by avinashtelsa** [Laravel-Backpack/community-forum#1226] [Q&A (Help)]
This is my code for saving and updating quotes in a JSON field, but it is being saved as follows:
```
protected function setupCreateOperation()
{
    Widget::add()->type('script')->content('js/admin/lesson_category_quotes.js');

    CRUD::setValidation(LessonCategoryQuotesRequest::class);
    CRUD::field('title');
    CRUD::field('course')
        ->type('select2')
        ->entity('course');
    CRUD::addField([
        'type'          => "relationship",
        'name'          => 'lessonCategory',
        'ajax'          => true,
        'minimum_input_length' => 0,
        'inline_create' => [ 'entity' => 'lesson-category' ] , 
        'data_source' => url('api/lesson-category'),
        'method' => 'GET',
    ]);
    CRUD::addField([
        'name'  => 'quotes',
        'label' => 'Quotes',
        'type'  => 'repeatable',
        'fields' => [
            [
                'name'  => 'title',
                'type'  => 'text',
                'label' => 'Quote Title',
            ],
            [
                'name'  => 'lessons',
                'type'  => 'select2_multiple',
                'label' => 'Lessons',
                'entity' => 'lessons',
                'model' => 'App\Models\Lesson',
                'attribute' => 'title',
                'pivot' => true,
                'default' => null,
                'placeholder' => 'Select lessons',
                'attributes' => [
                    'class' => 'lesson-select',
                ],
                'options' => (function ($query) {
                    return $query->where('lesson_category_id', 1)->get();
                }),
            ],
        ],
        'new_item_label'  => 'Add Quote',
        'init_rows'       => 0,
        'min_rows'        => 0,
        'max_rows'        => 10,
    ]);
}
```
```
[
  {
    "title": "Quote 1 Title",
    "lessons": [
      "1"
    ]
  },
  {
    "title": "Quote 2 Title",
    "lessons": [
      "2"
    ]
  }
]
```

But I want it to be saved like this:
```
[
    {
        "title": "Quote 1 Title",
        "lessons": [
            { "id": 1, "title": "Lesson 1" },
            { "id": 2, "title": "Lesson 2" }
        ]
    },
    {
        "title": "Quote 2 Title",
        "lessons": [
            { "id": 3, "title": "Lesson 3" }
        ]
    }
]
```

And if possible, I have statically passed the lesson_category_id here:
```
'options' => (function ($query) {
    return $query->where('lesson_category_id', 1)->get();
}),
```

I want to pass the selected value for this field here. Thank you for the help!

```
CRUD::addField([
    'type'               => "relationship",
    'name'               => 'lessonCategory',
    'ajax'               => true,
    'minimum_input_length' => 0,
    'inline_create'      => ['entity' => 'lesson-category'],
    'data_source'        => url('api/lesson-category'),
    'method'             => 'GET',
]);
```

[a comment by an automated account is not shown]

**Comment by avinashtelsa** [comment]
1) This answer to the previous question is correct.
```
    public function setQuotesAttribute($value)
    {
        // Check if $value is a string before decoding
        if (is_string($value)) {
            $quotes = json_decode($value, true);
        } else {
            $quotes = $value; // If $value is already an array, use it directly
        }

        foreach ($quotes as &$quote) {
            if (isset($quote['lessons'])) {
                $lessonIds = $quote['lessons'];
                $quote['lessons'] = Lesson::whereIn('id', $lessonIds)
                    ->get(['id', 'title'])
                    ->toArray();
            }
        }

        $this->attributes['quotes'] = json_encode($quotes);
    }
```

2) I need the selected value for this
```
CRUD::addField([
    'type'               => "relationship",
    'name'               => 'lessonCategory',
    'ajax'               => true,
    'minimum_input_length' => 0,
    'inline_create'      => [ 'entity' => 'lesson-category' ], // <--- THIS
    'data_source'        => url('api/lesson-category'),
    'method'             => 'GET',
    'attributes'         => [
        'id' => 'lesson-category',
    ],
]);
```

$lessonCategoryId = request()->input('lessonCategory');
Log::debug('lessonCategoryId: ' . $lessonCategoryId);


I tried this, but it didn’t work.

**Reply by jcastroa87** [reply]
Hello @avinashtelsa did you try to print all your request to check where did you receive the values?

```
dd(request());
```

If not see the value, please share the controller to replicate in my side and try to add the code style here to read it easily, thanks.

Cheers.

**Reply by avinashtelsa** [reply]
Hello @jcastroa87 

Using dd(request()) is not working. I need the value of the lessonCategory field at the time of adding or editing.

Here is my controller code.

protected function setupCreateOperation()
{
    Widget::add()->type('script')->content('js/admin/lesson_category_quotes.js');

    CRUD::setValidation(LessonCategoryQuotesRequest::class);
    CRUD::field('title');
    CRUD::field('course')
        ->type('select2')
        ->entity('course');
    CRUD::addField([
        'type'          => "relationship",
        'name'          => 'lessonCategory',
        'ajax'          => true,
        'minimum_input_length' => 0,
        'inline_create' => [ 'entity' => 'lesson-category' ] , // <--- THIS
        'data_source' => url('api/lesson-category'),
        'method' => 'GET',
        'attributes' => [
            'id' => 'lesson-category',
        ],
    ]);
    CRUD::addField([
        'name'    => 'character',
        'type'    => 'select2_from_array',
        'label'   => 'Character',
        'options' => [
            'Violence'               => 'Quote for Violence',
            'Ambition'               => 'Quote for Ambition',
            'Guilt'                  => 'Quote for Guilt',
            'Gender'                 => 'Quote for Gender',
            'Appearance vs Reality'  => 'Quote for Appearance vs Reality',
            'Supernatural'           => 'Quote for Supernatural',
            'Quotes 1-4'             => 'Quotes 1-4',
            'Quotes 5-6'             => 'Quotes 5-6',
            'Quotes 7-8'             => 'Quotes 7-8',
            'Quotes 1-6'             => 'Quotes 1-6',
        ],
        'default' => null,
        'placeholder' => 'Select any Character',
    ]);
    CRUD::addField([
        'name'  => 'quote_limits',
        'type'  => 'number',
        'label' => 'Quote Limit',
        'attributes' => [
            'min' => 1,
            'max' => 10,
        ],
        'hint' => 'Set the limit between 1 and 10',
        'default' => null,
    ]);
    CRUD::addField([
        'name'  => 'quotes',
        'label' => 'Quotes',
        'type'  => 'repeatable',
        'fields' => [
            [
                'name'  => 'title',
                'type'  => 'text',
                'label' => 'Quote Title',
            ],
            [
                'name'  => 'lessons',
                'type'  => 'select2_multiple',
                'label' => 'Lessons',
                'entity' => 'lessons',
                'model' => 'App\Models\Lesson',
                'attribute' => 'title',
                'pivot' => true,
                'default' => null,
                'placeholder' => 'Select lessons',
                'attributes' => [
                    'class' => 'lesson-select',
                ],
                'options' => function ($query) {
                    Log::info(print_r(request()->all(), true));
                    Log::debug('---------all---------');;

                    $lessonCategoryId = request()->input('lessonCategory');
                    if ($lessonCategoryId) {
                        return $query->where('lesson_category_id', $lessonCategoryId)->get();
                    }
                    return $query->get();
                },
            ],
        ],
        'new_item_label'  => 'Add Quote',
        'init_rows'       => 0,
        'min_rows'        => 0,
        'max_rows'        => 10,
    ]);

    /**
        * Fields can be defined using the fluent syntax:
        * - CRUD::field('price')->type('number');
        */
}

**Comment by jcastroa87** [comment]
Due to not activity, I will close this issue, but please feel free to re-open or create a new one.

Cheers.
